maestrowf/datastructures/yamlspecification.py

```diff
@@ -41,6 +41,42 @@
 
 logger = logging.getLogger(__name__)
 
+STUDY_STEP_KEYS = {
+    "name": None,
+    "description": None,
+    "run": {
+        "cmd": None,
+        "restart": None,
+        "depends": None,
+        "nodes": None,
+        "procs": None,
+        "cores per task": None,
+        "gpus per task": None,
+        "walltime": None,
+        "num resource set": None,
+    },
+}
+
+ENV_KEYS = {
+    "variables": None,
+    "labels": None,
+    "sources": None,
+    "dependencies": {
+        "path": {"name": None, "path": None},
+        "git": {"name": None, "path": None, "url": None},
+        "spack": {"name": None, "package_name": None},
+    },
+}
+
+def check_keys(parent_key, section, valid_keys):
```

Comment:
This could be a hidden method in the class itself. You could rework the API such that it fits within the abstract `Specification` class and is available to all specifications derived from it going forward.

Reply:
Good idea, I'll try that instead.